kronecker_test.py

```diff
@@ -0,0 +1,99 @@
+import numpy as np
+import tensorflow as tf
+
+import t3f
+import tensor_train
+import kronecker as kr
+import ops
+
+class KroneckerTest(tf.test.TestCase):
+
+  def testIsKronNonKron(self):
+    """
+    Tests _is_kron on a non-Kronecker matrix
+    """
+    tt_mat = t3f.get_variable('tt_mat', initializer=t3f.random_matrix(((2, 3), (3, 2)), tt_rank=2))
+    self.assertFalse(kr._is_kron(tt_mat))
+           
+  def testIsKronKron(self):
+    """
+    Tests _is_kron on a Kronecker matrix
+    """
+    kron_mat = t3f.get_variable('kron_mat', initializer=t3f.random_matrix(((2, 3), (3, 2)), tt_rank=1))
+    self.assertTrue(kr._is_kron(kron_mat))
+
+  def testDet(self):
+    """
+    Tests the determinant function
+    """
+    tf.set_random_seed(5)
+    kron_mat = t3f.get_variable('kron_mat', initializer=t3f.random_matrix(((2, 3, 2), (2, 3, 2)), tt_rank=1))
+    init_op = tf.global_variables_initializer()
+    with self.test_session() as sess:
+      sess.run(init_op)
+      desired = np.linalg.det(ops.full(kron_mat).eval())
+      actual = kr.determinant(kron_mat).eval()
+      self.assertAllClose(desired, actual)
+
+  def testSlogDet(self):
+    """
+    Tests the slog_determinant function
+    """
+    tf.set_random_seed(5) # negative derminant
+    kron_neg = t3f.get_variable('kron_neg', initializer=t3f.random_matrix(((2, 3), (2, 3)), tt_rank=1))
+  
+    tf.set_random_seed(1) # positive determinant
+    kron_pos = t3f.get_variable('kron_pos', initializer=t3f.random_matrix(((2, 3), (2, 3)), tt_rank=1))
```

Comment:
Add todo to use kron and -1 * kron when we implement __mul__ operation. The current version is dangerous -- random seed is not reliable across platforms, devices and future versions.